GoRogue/MapGeneration/RegionWithComponents.cs

```diff
@@ -0,0 +1,115 @@
+using System.Collections;
+using System.Collections.Generic;
+using GoRogue.Components;
+using GoRogue.Components.ParentAware;
+using SadRogue.Primitives;
+
+namespace GoRogue.MapGeneration
+{
+    /// <summary>
+    /// A Class which contains a Region, and a Component Collection.
+    /// </summary>
+    public class RegionWithComponents : IObjectWithComponents, IReadOnlyArea
+    {
+        /// <summary>
+        /// The region containing the components
+        /// </summary>
+        public Region Region { get; private set; }
+
+        /// <summary>
+        /// The GoRogueComponents on this region
+        /// </summary>
+        public IComponentCollection GoRogueComponents { get; set; }
```

Comment:
Recommend _not_ allowing assignment after construction on this if possible, as doing so carries notable risk of introducing parent desyncs to the component system:
```suggestion
        public IComponentCollection GoRogueComponents { get; }
```